File: LeetCode75-sols/Array_String/JavaScript/Merge_Strings_Alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let result = "";
    let i = 0;
    let len1 = word1.length;
    let len2 = word2.length;

    while ((i < len1) || (i < len2)) {
        if (i < len1) {
            result += word1[i];
        }
        if (i < len2) {
            result += word2[i];
        }
        i++;
    }
    return result;
};

File: LeetCode75-sols/Array_String/C#/Merge_Strings_Alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        var result = new System.Text.StringBuilder();
        int i = 0;
        int len1 = word1.Length;
        int len2 = word2.Length;

        while ((i < len1) || (i < len2)) {
            if (i < len1) {
                result.Append(word1[i]);
            }
            if (i < len2) {
                result.Append(word2[i]);
            }
            i++;
        }
        return result.ToString();
    }
}
